cobbler/cobbler_collections/collection.py

```diff
@@ -155,10 +155,65 @@ def remove(
         :param with_triggers: run "on delete" triggers
         :param recursive: recursively delete children
         :param rebuild_menu: rebuild menu after removing the item
-        :returns: NotImplementedError
+        :raises CX: In case any subitem (images, profiles or systems) would be orphaned. If the option ``recursive``
+                    is set then the orphaned items would be removed automatically.
         """
+        if ref is None:  # type: ignore
+            raise CX("cannot delete an object that does not exist")
+
+        for item_type in InheritableItem.TYPE_DEPENDENCIES[ref.COLLECTION_TYPE]:
+            dep_type_items = self.api.find_items(
+                item_type.dependant_item_type,
+                {item_type.dependant_type_attribute: ref.uid},
+                return_list=True,
+            )
+            if dep_type_items is None or not isinstance(dep_type_items, list):
+                raise ValueError("Expected list to be returned by find_items")
+            if len(dep_type_items) > 0:
+                if recursive:
+                    for dep_item in dep_type_items:
+                        self.api.remove_item(
+                            dep_item.COLLECTION_TYPE,
+                            dep_item,
+                            recursive=recursive,
+                            delete=with_delete,
+                            with_triggers=with_triggers,
+                            with_sync=with_sync,
+                        )
+                else:
+                    dep_str = ",".join([ref.uid for ref in dep_type_items])
+                    raise CX(
+                        f"removal would orphan {ref.COLLECTION_TYPE}(s): {dep_str}"
```

Comment:
I am not sure, but could it be that using `ref` is incorrect here? Maybe I am overlooking something...

```suggestion
                        f"removal would orphan {item_type.dependant_item_type.COLLECTION_TYPE}(s): {dep_str}"
```